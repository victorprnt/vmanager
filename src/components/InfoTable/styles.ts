import styled from 'styled-components'

export const Wrapper = styled.section`
  display: flex;
  width: 100%;
  height: 100%;

  .table-container {
    width: 100%;
    max-height: 100%;
    margin: 3vw 3vw 3vw 6vw;

    padding: 2rem;
    background-color: #fff;
    color: #000;

    table {
      max-width: 100%;
    }
  }
`
