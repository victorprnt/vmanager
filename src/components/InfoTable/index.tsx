import TableRow from '../TableRow'
import * as S from './styles'

const InfoTable = () => (
  <S.Wrapper>
    <div className="table-container">
      <div className="input">
        <input type="text" className="filter" placeholder="Search Filter" />
        <button type="button" className="search-button">
          Search
        </button>
      </div>

      <table>
        <thead>
          <tr>
            <th>Column 01</th>
            <th>Column 02</th>
            <th>Column 03</th>
            <th>Column 04</th>
            <th>Column 05</th>
            <th>Column 06</th>
            <th>Column 07</th>
          </tr>
        </thead>
        <tbody>
          <TableRow />
          <TableRow />
          <TableRow />
          <TableRow />
        </tbody>
      </table>

      <div className="footer">
        <p className="text-results">Showing 1-50 of 200 results</p>
        <div className="page-selector"> {'<< < 1 2 > >>'} </div>
      </div>
    </div>
  </S.Wrapper>
)

export default InfoTable
