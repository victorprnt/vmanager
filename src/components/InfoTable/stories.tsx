// import { Story, Meta } from '@storybook/react'
import InfoTable from '.'

// export default {
//   title: 'InfoTable',
//   component: InfoTable
// } as Meta

// export const Default: Story = () => <InfoTable />
