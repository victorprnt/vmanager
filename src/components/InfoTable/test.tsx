import { render, screen } from '@testing-library/react'

import InfoTable from '.'

describe('<InfoTable />', () => {
  it('should render the table content', () => {
    const { container } = render(<InfoTable />)

    expect(screen.getByRole('textbox')).toHaveAttribute(
      'placeholder',
      'Search Filter'
    )
    expect(screen.getByRole('table')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
