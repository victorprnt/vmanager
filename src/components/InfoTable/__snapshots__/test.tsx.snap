// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoTable /> should render the table content 1`] = `
<section
  class="sc-gsnTZi hsaFmk"
>
  <div
    class="table-container"
  >
    <div
      class="input"
    >
      <input
        class="filter"
        placeholder="Search Filter"
        type="text"
      />
      <button
        class="search-button"
        type="button"
      >
        Search
      </button>
    </div>
    <table>
      <thead>
        <tr>
          <th>
            Column 01
          </th>
          <th>
            Column 02
          </th>
          <th>
            Column 03
          </th>
          <th>
            Column 04
          </th>
          <th>
            Column 05
          </th>
          <th>
            Column 06
          </th>
          <th>
            Column 07
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="sc-bczRLJ"
        >
          <td>
            Info 01
          </td>
          <td>
            Info 02
          </td>
          <td>
            Inof 03
          </td>
          <td>
            Info 04
          </td>
          <td>
            Info 05
          </td>
          <td>
            Info 06
          </td>
          <td>
            Info 07
          </td>
        </tr>
        <tr
          class="sc-bczRLJ"
        >
          <td>
            Info 01
          </td>
          <td>
            Info 02
          </td>
          <td>
            Inof 03
          </td>
          <td>
            Info 04
          </td>
          <td>
            Info 05
          </td>
          <td>
            Info 06
          </td>
          <td>
            Info 07
          </td>
        </tr>
        <tr
          class="sc-bczRLJ"
        >
          <td>
            Info 01
          </td>
          <td>
            Info 02
          </td>
          <td>
            Inof 03
          </td>
          <td>
            Info 04
          </td>
          <td>
            Info 05
          </td>
          <td>
            Info 06
          </td>
          <td>
            Info 07
          </td>
        </tr>
        <tr
          class="sc-bczRLJ"
        >
          <td>
            Info 01
          </td>
          <td>
            Info 02
          </td>
          <td>
            Inof 03
          </td>
          <td>
            Info 04
          </td>
          <td>
            Info 05
          </td>
          <td>
            Info 06
          </td>
          <td>
            Info 07
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="footer"
    >
      <p
        class="text-results"
      >
        Showing 1-50 of 200 results
      </p>
      <div
        class="page-selector"
      >
         
        &lt;&lt; &lt; 1 2 &gt; &gt;&gt;
         
      </div>
    </div>
  </div>
</section>
`;
